File: app/src/main/java/com/openclassrooms/realestatemanager/PhotoRepository.java
package com.openclassrooms.realestatemanager;

import androidx.lifecycle.LiveData;

import java.util.List;

public class PhotoRepository {

    private final PhotoDao photoDao;

    public PhotoRepository(PhotoDao photoDao) {
        this.photoDao = photoDao;
    }

    public LiveData<List<Photo>> getPhotos(long realEstateId) { return this.photoDao.getAllPhotos(realEstateId); }

    public void createPhoto(Photo photo) { photoDao.insert(photo); }
    public void updatePhoto(Photo photo) { photoDao.update(photo); }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/repository/PhotoRepository.kt
package com.openclassrooms.realestatemanager.repository

import androidx.lifecycle.LiveData
import com.openclassrooms.realestatemanager.dao.PhotoDao
import com.openclassrooms.realestatemanager.model.Photo

class PhotoRepository {

    lateinit var photoDao: PhotoDao

    fun getPhotos(realEstateId:Int):LiveData<List<Photo>> { return this.photoDao.getPhotos(realEstateId) }
    suspend fun insertPhoto(photo: Photo) { photoDao.insertPhoto(photo) }
    suspend fun deletePhoto(photoId:Int) { photoDao.deletePhoto(photoId) }
    suspend fun updatePhoto(photo:Photo) { photoDao.updatePhoto(photo) }
}